src/components/interactive-ad.js

```diff
@@ -89,24 +89,19 @@ export class InteractiveAd {
         }
 
         async function getNativePlatformAdvertisingId() {
-            if (window.fireTvApp) {
-                if (!window.webApp) {
-                    window.webApp = {};
-                }
-
+            if (window.webApp && window.fireTvApp) {
                 return new Promise(function(resolve, reject) {
-                    window.webApp.onAdvertisingIdReady = function(advertisingId, isLimitAdTrackingEnabled) {
-                        resolve(arguments.length == 0 ? null: advertisingId);
+                    window.webApp.onAdvertisingIdReady = function(advertisingId) {
+                        // consume the callback
+                        window.webApp.onAdvertisingIdReady = null;
+                        resolve(advertisingId);
                     }
 
                     window.fireTvApp.getAdvertisingId && window.fireTvApp.getAdvertisingId();
                 });
             } else {
                 return Promise.resolve(undefined);
             }
-
-            // TODO: use true platform specific advertising id, e.g. bridge call to native side for FireTV
-            return undefined;
```

Comment:
keep the return statement at least, to indicate we are explicit returning nothing on purpose as a fallback